design/spec/upsert/tests.md

```
@@ -0,0 +1,120 @@
+> Test plan for `upsert` (#847): 39 test cases covering unit, concurrency, type coverage, erase, contract verification, and OOM.
```

markdownlint-cli2 (0.22.1)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)